extension/builtin/default_registry.go

```diff
@@ -36,62 +29,20 @@ var (
 	MySQLURIFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "uri to use with mysql storage")
 )
 
-// Default implementation of extension.Registry.
-type defaultRegistry struct {
-	db *sql.DB
+// UseSQLDB configures the registry to use the given SQL database for all storage.
+func UseSQLDB(r *extension.Registry, db *sql.DB) {
```

Comment:
Would it be possible to convert these `Use*` functions to `New*` functions? 
For example, this function modifies the given registry which could be surprising for the caller.
Could this be a NewFromDB which returned a new registry that's populated in the way you suggest here?